src/Yarhl/IO/DataStream.cs

```diff
@@ -46,7 +48,11 @@ public class DataStream : IDisposable
         /// </summary>
         public DataStream()
         {
-            BaseStream = new RecyclableMemoryStream();
+            BaseStream = new RecyclableMemoryStream
```

Comment:
Can you move the initialization of the `LockObj` to the default constructor of `RecyclableMemoryStream`? And check if there are other implementations of `IStream` missing the initialization?

Otherwise we would need to add it to the Factory too and make sure user initialize it.